zest/releaser/hg.py

```diff
@@ -21,14 +24,17 @@ def name(self):
         # No setup.py? With hg we can probably only fall back to the directory
         # name as there's no svn-url with a usable name in it.
         dir_name = os.path.basename(os.getcwd())
+        if not isinstance(dir_name, six.text_type):
+            dir_name = dir_name.decode(sys.getfilesystemencoding())
         return dir_name
 
     def available_tags(self):
-        tag_info = execute_command('hg tags')
-        tags = [line[:line.find(' ')] for line in tag_info.split('\n')]
+        tag_info = execute_command([u'hg', u'tags'])
+        tags = [line[:line.find(u' ')] for line in tag_info.split(u'\n')]
         tags = [tag for tag in tags if tag]
-        tags.remove('tip')  # Not functional for us
-        logger.debug("Available tags: %r", tags)
+        while u'tip' in tags:
+            tags.remove(u'tip')  # Not functional for us
```

Comment:
Now I'm curious about the `while`.

(Suggest `tags = [tag for tag in tags if tag and tag != u'tip']` instead.)